src/Kiota.Builder/Writers/Go/CodeEnumWriter.cs

```diff
@@ -28,20 +28,29 @@ public override void WriteCodeElement(CodeEnum codeElement, LanguageWriter write
                         string.Empty,
                         "const (");
         writer.IncreaseIndent();
+        var isMultiValue = codeElement.Flags;
+
         var iotaSuffix = $" {typeName} = iota";
         var enumOptions = codeElement.Options;
+        int power = 0;
         foreach (var item in enumOptions)
         {
             if (!string.IsNullOrEmpty(item.Documentation.Description))
                 writer.WriteLine($"// {item.Documentation.Description}");
-            writer.WriteLine($"{item.Name.ToUpperInvariant()}_{typeName.ToUpperInvariant()}{iotaSuffix}");
+
+            if (isMultiValue)
+                writer.WriteLine($"{item.Name.ToUpperInvariant()}_{typeName.ToUpperInvariant()} = {(int)Math.Pow(2, power)}");
+            else
+                writer.WriteLine($"{item.Name.ToUpperInvariant()}_{typeName.ToUpperInvariant()}{iotaSuffix}");
+
             if (!string.IsNullOrEmpty(iotaSuffix))
                 iotaSuffix = string.Empty;
```

Comment:
this could be done in the previous else block to avoid assigning strings when we don't need to